uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/apis/leave-request-api.js

```diff
@@ -55,24 +57,35 @@ define([
           public_holiday: isPublicHoliday || false
         };
 
-        deferred.resolve(this.sendGET('LeaveRequest', 'getbalancechangebyabsencetype', params)
+        return this.sendGET('LeaveRequest', 'getbalancechangebyabsencetype', params)
           .then(function (data) {
             return data.values;
-          }));
-
-        return deferred.promise;
+          });
       },
 
       /**
        * This method is used to update a leave request
        *
-       * @param {object} updatedAttributes - Updated values of leave request
+       * @param {object} params - Updated values of leave request
        * @return {Promise} Resolved with {Object} Updated Leave request
        */
-      update: function (updatedAttributes) {
+      update: function (params) {
         $log.debug('LeaveRequestAPI.update');
+        var deferred = $q.defer();
 
-        return this.sendPOST('LeaveRequest', 'create', updatedAttributes);
+        if (!('id' in params)) {
+          deferred.reject({
+            is_error: 1,
+            error_message: 'id is mandatory field'
+          });
+          return deferred.promise;
+        }
+
+        return this.sendPOST('LeaveRequest', 'create', params)
+          .then(function (data) {
+            //returns array of single object hence getting first object
```

Comment:
Have you tried passing `sequential: 0` in the `params`? `sequential` is used by the API to determine whether `values` should be an array or an object. In the base `api` service it's set to `1`, but you can always override it if you need to do so